File: frontend/wechat-prototype/src/services/resumes.js
import { get, post, postForm, put } from './api.js'

export async function getMyResume() {
  return get('/api/v1/resumes/me')
}

export async function uploadResume(file) {
  const formData = new FormData()
  formData.append('file', file)
  const result = await postForm('/api/v1/resumes/me/upload', formData)
  const uploadFailed = result?.upload?.status === 'failed'
  const parseFailed = result?.parse_run?.status && result.parse_run.status !== 'succeeded'
  if (uploadFailed || parseFailed) {
    throw new Error(result?.upload?.error_message || result?.parse_run?.error_message || '简历解析失败，请换一份文件重试')
  }
  if (result?.resume) {
    return {
      ...result.resume,
      upload: result.upload,
      parse_run: result.parse_run,
    }
  }
  return result
}

// R-P1-06 联调：上传历史列表
// 返回 [{ upload, latest_parse_run }]
export async function listMyUploads(limit = 20) {
  return get(`/api/v1/resumes/me/uploads?limit=${limit}`)
}

// R-P1-06 联调：解析详情（含原文预览 + chunks）
// 返回 { upload, parse_run, extracted_text, chunks }
export async function getParseRunDetail(parseRunId) {
  return get(`/api/v1/resumes/me/parse-runs/${parseRunId}`)
}

// R-P2-04：获取解析结果结构化数据
// 返回 { basic, educations, work_experiences, projects, skills, certificates }
export async function getStructuredResult(parseRunId) {
  return get(`/api/v1/resumes/me/parse-runs/${parseRunId}/structured`)
}

// R-P2-04：确认解析结果并投影到明细表
export async function confirmStructuredResult(parseRunId, data) {
  return put('/api/v1/resumes/me/structured/confirm', {
    parse_run_id: parseRunId,
    ...data
  })
}

// R-P2-05：获取画像聚合数据
export async function getProfileSummary() {
  return get('/api/v1/resumes/me/profile-summary')
}

export async function getRecruiterApplicationStructuredProfile(applicationId) {
  return get(`/api/v1/resumes/recruiter/applications/${applicationId}/structured-profile`)
}

File: backend/job-platform/app/api/v1/resumes.py
"""
Seeker resume API.
"""
from fastapi import APIRouter, Depends, File, Query, UploadFile
from sqlalchemy.ext.asyncio import AsyncSession

from app.core.dependencies import require_role
from app.db.session import get_db
from app.modules.resume.schemas import (
    ResumeParseRunDetailResponse,
    ResumeProfileSummaryResponse,
    ResumeStatusResponse,
    ResumeStructuredConfirmRequest,
    ResumeStructuredProfileCreateRequest,
    ResumeStructuredProfileDetailResponse,
    ResumeStructuredProfileResponse,
    ResumeStructuredProjectionRequest,
    ResumeStructuredProjectionResponse,
    ResumeUploadHistoryItemResponse,
    ResumeUploadResultResponse,
)
from app.modules.resume.service import ResumeService
from app.modules.user.models import User

router = APIRouter()


@router.get("/me", response_model=ResumeStatusResponse)
async def get_my_resume(
    current_user: User = Depends(require_role("seeker")),
    db: AsyncSession = Depends(get_db),
):
    """Get current seeker's latest uploaded resume."""
    return await ResumeService.get_my_status(db, current_user)


@router.get("/me/profile-summary", response_model=ResumeProfileSummaryResponse)
async def get_my_resume_profile_summary(
    current_user: User = Depends(require_role("seeker")),
    db: AsyncSession = Depends(get_db),
):
    """Get current seeker's aggregated resume profile for portrait UI."""
    return await ResumeService.get_my_profile_summary(db, current_user)


@router.get("/me/uploads", response_model=list[ResumeUploadHistoryItemResponse])
async def list_my_resume_uploads(
    limit: int = Query(20, ge=1, le=50),
    current_user: User = Depends(require_role("seeker")),
    db: AsyncSession = Depends(get_db),
):
    """List current seeker's resume upload history."""
    return await ResumeService.list_my_uploads(db, current_user, limit=limit)


@router.get("/me/parse-runs/{parse_run_id}", response_model=ResumeParseRunDetailResponse)
async def get_my_parse_run_detail(
    parse_run_id: int,
    current_user: User = Depends(require_role("seeker")),
    db: AsyncSession = Depends(get_db),
):
    """Get parse run detail for current seeker's resume preview."""
    return await ResumeService.get_my_parse_run_detail(db, current_user, parse_run_id)


@router.get("/me/parse-runs/{parse_run_id}/structured", response_model=ResumeStructuredProfileDetailResponse)
async def get_my_parse_run_structured_profile(
    parse_run_id: int,
    current_user: User = Depends(require_role("seeker")),
    db: AsyncSession = Depends(get_db),
):
    """Get structured profile for one parse run."""
    return await ResumeService.get_my_structured_profile_by_parse_run(db, current_user, parse_run_id)


@router.put("/me/structured/confirm", response_model=ResumeStructuredProjectionResponse)
async def confirm_my_structured_profile(
    payload: ResumeStructuredConfirmRequest,
    current_user: User = Depends(require_role("seeker")),
    db: AsyncSession = Depends(get_db),
):
    """Confirm structured JSON by parse run and project it to detail tables."""
    return await ResumeService.confirm_structured_profile_by_parse_run(db, current_user, payload)


@router.post("/me/structured/confirm", response_model=ResumeStructuredProjectionResponse)
async def confirm_my_structured_profile_compat(
    payload: ResumeStructuredConfirmRequest,
    current_user: User = Depends(require_role("seeker")),
    db: AsyncSession = Depends(get_db),
):
    """Backward-compatible confirm endpoint for older clients that still POST."""
    return await ResumeService.confirm_structured_profile_by_parse_run(db, current_user, payload)


@router.get("/me/structured-profiles/latest", response_model=ResumeStructuredProfileDetailResponse)
async def get_my_latest_structured_profile(
    current_user: User = Depends(require_role("seeker")),
    db: AsyncSession = Depends(get_db),
):
    """Get current seeker's latest structured resume profile."""
    return await ResumeService.get_my_latest_structured_profile(db, current_user)


@router.get("/recruiter/applications/{application_id}/structured-profile", response_model=ResumeStructuredProfileDetailResponse)
async def get_recruiter_application_structured_profile(
    application_id: int,
    current_user: User = Depends(require_role("recruiter")),
    db: AsyncSession = Depends(get_db),
):
    """Get structured resume profile for one application owned by current recruiter."""
    return await ResumeService.get_recruiter_application_structured_profile(db, current_user, application_id)


@router.get("/me/structured-profiles/{profile_id}", response_model=ResumeStructuredProfileDetailResponse)
async def get_my_structured_profile(
    profile_id: int,
    current_user: User = Depends(require_role("seeker")),
    db: AsyncSession = Depends(get_db),
):
    """Get one structured profile and normalized projection rows."""
    return await ResumeService.get_my_structured_profile_detail(db, current_user, profile_id)


@router.post("/me/structured-profiles", response_model=ResumeStructuredProfileResponse)
async def create_my_structured_profile(
    payload: ResumeStructuredProfileCreateRequest,
    current_user: User = Depends(require_role("seeker")),
    db: AsyncSession = Depends(get_db),
):
    """Create or replace structured JSON for one parse run."""
    return await ResumeService.create_structured_profile(db, current_user, payload)


@router.post(
    "/me/structured-profiles/{profile_id}/project",
    response_model=ResumeStructuredProjectionResponse,
)
async def project_my_structured_profile(
    profile_id: int,
    payload: ResumeStructuredProjectionRequest,
    current_user: User = Depends(require_role("seeker")),
    db: AsyncSession = Depends(get_db),
):
    """Project structured JSON into normalized resume detail tables."""
    return await ResumeService.project_structured_profile(db, current_user, profile_id, payload)


@router.post("/me/upload", response_model=ResumeUploadResultResponse)
async def upload_my_resume(
    file: UploadFile = File(...),
    current_user: User = Depends(require_role("seeker")),
    db: AsyncSession = Depends(get_db),
):
    """Upload or replace current seeker's resume."""
    return await ResumeService.upload(db, current_user, file)
